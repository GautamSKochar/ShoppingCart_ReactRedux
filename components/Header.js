import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
var cartStyle = {
    'whiteSpace': 'nowrap',
    'textAlign': 'center',
    'lineHeight': '18px',
    'padding': '0 5px',
    'height': '18px',
    'background': '#007bff',
    'position': 'static',
    'borderRadius': '5px'
}

class Header extends React.Component {
    render() {
        return (

            <div className="page-header">

                <nav className="nav navbar navbar-expand-lg navbar-dark bg-dark navbar-fixed-top">
                    <div className="container-fluid">
                        <ul className="nav navbar-nav">
                            <li><NavLink to="/">eShop</NavLink></li>
                        </ul>
                        <h1 style={{ align: "center", color: 'white' }}>Shopping cart</h1>
                        <ul className="nav navbar-nav navbar-right">
                            <li><NavLink to="/cart">Cart&nbsp;
                            {this.props.cartCount ? <span style={cartStyle}>
                                    {this.props.cartCount}
                                </span> : ''}

                            </NavLink></li>
                        </ul>
                    </div>
                </nav>
            </div >



        );
    }
}

const mapStateToProps = (state) => {
    return {
        cartCount: state.eShopReducer.cart ? state.eShopReducer.cart.length : 0
    }
}
export default connect(mapStateToProps)(Header)