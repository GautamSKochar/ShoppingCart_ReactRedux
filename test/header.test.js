import React from "react";
import Header from "../components/Header";
import { shallow , mount, render} from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Provider} from "react-redux";
import configureMockStore from "redux-mock-store";
import toJson from "enzyme-to-json";


const raf = global.requestAnimationFrame = (cb) => {
    setTimeout(cb, 0)
}
configure({ adapter: new Adapter() });
const store = configureMockStore({});

describe('Header Component', () => {
    it('Snapshot Test', () => {
        const component = shallow(<Provider store={store}><Header /></Provider>);
        const tree = toJson(component);
        expect(tree).toMatchSnapshot();
    })
});