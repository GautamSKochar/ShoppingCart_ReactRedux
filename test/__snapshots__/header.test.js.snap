// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Snapshot Test 1`] = `<Connect(Header) />`;
